### PR TITLE
fix: add redis tracker store ssl param

### PR DESCRIPTION
**Pull request by mbelang**
**Proposed changes**:
- Some deployment of redis has transport encryption at rest e.g. Redis Elasticache on AWS. That feature is a must have when speaking of production deployment.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by msamogh** [comment]
Thanks for the PR, we'll give it a review as soon as possible.

File: tests/core/test_tracker_stores.py [hunk 1]
```diff
@@ -92,6 +92,25 @@ def test_find_tracker_store(default_domain):
     )
 
 
+def test_find_redis_tracker_store_with_ssl(default_domain):
```
Comment (wochinge) [inline tests/core/test_tracker_stores.py]: I don't think this test is testing anything which wasn't cover before, so I we could just skip this test in my opinion.

File: data/test_endpoints/example_endpoints.yml [hunk 2]
```diff
@@ -22,4 +22,3 @@ tracker_store:
   #db: rasa
   #user: username
   #password: password
-
```
Comment (wochinge) [inline data/test_endpoints/example_endpoints.yml]: please keep the empty line at the end

**Comment by tmbo** [comment]
thanks a lot for this PR, @mbelang do you mind taking a look at the comments so we can get this merged?

**Comment by wochinge** [comment]
@mbelang Do you have time to work in the changes? We would really appreciate it 🎉

File: rasa/core/tracker_store.py [hunk 3]
```diff
@@ -172,11 +172,14 @@ def __init__(
...
         event_broker=None,
         record_exp=None,
+        ssl=False,
     ):
 
         import redis
 
-        self.red = redis.StrictRedis(host=host, port=port, db=db, password=password)
+        self.red = redis.StrictRedis(
+            host=host, port=port, db=db, password=password, ssl=ssl
```
[the remark this answers is not in the document]
Comment (mbelang) [reply rasa/core/tracker_store.py]: I do not think so. I did implement the same fix on a custom implementation of tracker store and was not required.

**Comment by akelad** [comment]
@mbelang i'm sorry that i'm hijacking your PR but it's a contribution we need to patch 1.2.x with as well. I've cherry picked all your commits so the history isn't lost: https://github.com/RasaHQ/rasa/pull/4615

**Comment by mbelang** [comment]
@akelad No worries :) Happy to see that merged. FYI, you will need to do same fix for the lock store :)

**Comment by akelad** [comment]
AH good point! @ricwo could you take that on?

**Pull request closed without merging**